service.subtitles.wizdom/service.py

```diff
@@ -1,297 +1,345 @@
+		item['file_original_path'] = ""
+		labelType = getInfoLabel("ListItem.DBTYPE")  # movie/tvshow/season/episode
+		isItMovie = labelType == 'movie' or getCondVisibility("Container.Content(movies)")
+		isItEpisode = labelType == 'episode' or getCondVisibility("Container.Content(episodes)")
+
+		if isItMovie:
+			item['title'] = getInfoLabel("ListItem.OriginalTitle")
+		elif isItEpisode:
+			item['title'] = getInfoLabel("ListItem.TVShowTitle")
+		else:
+			item['title'] = "SearchFor..."  # In order to show "No Subtitles Found" result.
+	
+	wlog("item:%s" % item)
+	imdb_id = 0
+	try:
+		if Player().isPlaying():	# Enable using subtitles search dialog when kodi is not playing
+			playerid_query = '{"jsonrpc": "2.0", "method": "Player.GetActivePlayers", "id": 1}'
+			playerid = loads(executeJSONRPC(playerid_query))['result'][0]['playerid']
+			imdb_id_query = '{"jsonrpc": "2.0", "method": "Player.GetItem", "params": {"playerid": ' + \
+				str(playerid) + ', "properties": ["imdbnumber"]}, "id": 1}'
+			imdb_id = loads(executeJSONRPC(imdb_id_query))['result']['item']['imdbnumber']
+			wlog("imdb JSONPC:%s" % imdb_id)
+		else:
+			if labelIMDB:
+				imdb_id = labelIMDB
+			else:
+				if isItMovie:
+					imdb_id = "ThisIsMovie"  # Search the movie by item['title'] for imdb_id
+				elif isItEpisode:
+					imdb_id = "ThisIsEpisode"  # Search by item['title'] for tvdb_id
+				else:
+					imdb_id = "tt0"  # In order to show "No Subtitles Found" result => Doesn't recognize movie/episode
+	except Exception as err:
+		wlog('Caught Exception: error in imdb id: %s' % format(err))		
+		pass
+
+	if imdb_id[:2] == "tt":  # Simple IMDB_ID
+		searchByIMDB(imdb_id, item['season'], item['episode'], item['file_original_path'])
+	else:
+		# Search TV Show by Title
+		if item['season'] or item['episode']:
+			try:
+				imdb_id = searchTMDB("tv",quote(item['title']),0)
+				wlog("Search TV IMDB:%s [%s]" % (imdb_id, item['title']))
+				if imdb_id[:2] == "tt":
+					searchByIMDB(imdb_id, item['season'], item['episode'], item['file_original_path'])
+			except Exception as err:
+				wlog('Caught Exception: error in tv search: %s' % format(err))
+				pass
+		# Search Movie by Title+Year
+		else:
+			try:
+				imdb_id = searchTMDB("movie",query=item['title'], year=item['year'])
+				wlog("Search IMDB:%s" % imdb_id)
+				if not imdb_id[:2] == "tt":
+					imdb_id = searchTMDB("movie",query=item['title'], year=(int(item['year'])-1))
+					wlog("Search IMDB(2):%s" % imdb_id)
+				if imdb_id[:2] == "tt":
+					searchByIMDB(imdb_id, 0, 0, item['file_original_path'])
+			except Exception as err:
+				wlog('Caught Exception: error in movie search: %s' % format(err))
+				pass
+
+	# Search Local File
+	if not imdb_id:
+		ManualSearch(item['title'])
+	endOfDirectory(int(sys.argv[1]))
+	if myAddon.getSetting("Debug") == "true":
+		if imdb_id[:2] == "tt":
+			Dialog().ok("Debug "+myVersion, str(item), "imdb: "+str(imdb_id))
+		else:
+			Dialog().ok("Debug "+myVersion, str(item), "NO IDS")
```

Comment:
NO IDS string needs to be localized as it is presented to the user. Same goes for the "Debug" string.